libdevice/nativecpu_utils.cpp

```diff
@@ -329,26 +329,23 @@ DefineShuffleVec2to16(float, f32, float);
   DEVICE_EXTERNAL GET_PROPS uint32_t bname() { return muxname(); }             \
   static_assert(true)
 // subgroup
-GEN_u32(__spirv_SubgroupLocalInvocationId, __mux_get_sub_group_local_id);
-GEN_u32(__spirv_SubgroupMaxSize, __mux_get_max_sub_group_size);
-GEN_u32(__spirv_SubgroupId, __mux_get_sub_group_id);
+GEN_u32(__spirv_BuiltInSubgroupLocalInvocationId, __mux_get_sub_group_local_id);
+GEN_u32(__spirv_BuiltInSubgroupMaxSize, __mux_get_max_sub_group_size);
+GEN_u32(__spirv_BuiltInSubgroupId, __mux_get_sub_group_id);
 
 // I64_I32
-#define GEN_p(bname, muxname, arg)                                             \
+#define GEN_p(bname, muxname)                                                  \
   DEVICE_EXTERN_C GET_PROPS uint64_t muxname(uint32_t);                        \
-  DEVICE_EXTERNAL GET_PROPS uint64_t bname() { return muxname(arg); }          \
+  DEVICE_EXTERNAL GET_PROPS uint64_t bname(int dim) { return muxname(dim); }   \
   static_assert(true)
 
-#define GEN_xyz(bname, ncpu_name)                                              \
-  GEN_p(bname##_x, ncpu_name, 0);                                              \
-  GEN_p(bname##_y, ncpu_name, 1);                                              \
-  GEN_p(bname##_z, ncpu_name, 2)
+#define GEN_xyz(bname, ncpu_name) GEN_p(bname, ncpu_name);
```

Comment:
All these macros were defined to not have a trailing comma at the end to avoid empty declarations when the point of use adds another semicolon.

Now that `GEN_p` and `GEN_xyz` do the same thing, we don't need two macros, can you remove `GEN_xyz` and then rename `GEN_p` to `GEN_xyz`?

Reply:
> All these macros were defined to not have a trailing comma at the end to avoid empty declarations when the point of use adds another semicolon.
> 
> Now that `GEN_p` and `GEN_xyz` do the same thing, we don't need two macros, can you remove `GEN_xyz` and then rename `GEN_p` to `GEN_xyz`?

done, thanks